File: app/api/routes/knowledge_base.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.api.deps import get_db_session
from app.schemas.knowledge_base import (
    KnowledgeBaseOverview,
    PolicyDocumentOptionList,
    RagMvpStatus,
)
from app.services.knowledge_base import KnowledgeBaseService

router = APIRouter()


@router.get("/overview", response_model=KnowledgeBaseOverview)
async def get_knowledge_base_overview() -> KnowledgeBaseOverview:
    return KnowledgeBaseService().get_overview()


@router.get("/mvp-status", response_model=RagMvpStatus)
async def get_rag_mvp_status() -> RagMvpStatus:
    return KnowledgeBaseService().get_rag_mvp_status()


@router.get("/documents", response_model=PolicyDocumentOptionList)
async def list_policy_documents(
    search: str | None = Query(default=None, description="按制度名称模糊搜索。"),
    policy_category: str | None = Query(default=None, description="按制度分类过滤。"),
    limit: int = Query(default=50, ge=1, le=200, description="返回数量上限。"),
    session: Session = Depends(get_db_session),
) -> PolicyDocumentOptionList:
    return KnowledgeBaseService().list_documents(
        session,
        search=search,
        policy_category=policy_category,
        limit=limit,
    )

File: frontend/src/api/knowledgeBase.ts
import type { PolicyDocumentOptionList } from "../types/knowledgeBase";

const API_BASE = "/api/v1/kb";

export async function listPolicyDocuments(params?: {
  search?: string;
  policyCategory?: string;
  limit?: number;
}): Promise<PolicyDocumentOptionList> {
  const query = new URLSearchParams();
  if (params?.search?.trim()) {
    query.set("search", params.search.trim());
  }
  if (params?.policyCategory?.trim()) {
    query.set("policy_category", params.policyCategory.trim());
  }
  if (params?.limit) {
    query.set("limit", String(params.limit));
  }

  const response = await fetch(
    `${API_BASE}/documents${query.size ? `?${query.toString()}` : ""}`,
  );

  if (!response.ok) {
    throw new Error(`请求失败: ${response.status}`);
  }

  return (await response.json()) as PolicyDocumentOptionList;
}
